### PR TITLE
shipit.config should be prioritized for remoteCopy command.

### PR DESCRIPTION
**Pull request by mochja**
Based on #83.

### CODE REVIEW

File: lib/shipit.js [hunk 1]
```diff
@@ -240,9 +240,10 @@ Shipit.prototype.remoteCopy = function (src, dest, options, callback) {
     options = undefined;
   }
 
-  options = _.defaults(options || {}, {
-    ignores: this.config && this.config.ignores ? this.config.ignores : [],
-    rsync: this.config && this.config.rsync ? this.config.rsync : []
+  options = options || {};
+  options = _.assign(options, {
```
Comment (neoziro) [inline lib/shipit.js]: To avoid altering the `options` object:
```js
_.assign({}, options, {...});
```

**Comment by neoziro** [comment]
@mochja just a small fix and I merge it

**Comment by mochja** [comment]
@neoziro :shipit:

**Comment by grEvenX** [comment]
Seems like bad practice to have global options override method-specific options?
The feature to allow it to use `shipit.config` is good, but if you actually specify the options when calling the `remoteCopy` method I would expect them to override any values set in a global config.

If you agree, the test should be changed to the following expectations: ?
```
it('should support options specified in shipit.config', function () {
  shipit.config = {
    ignores: ['foo'],
    rsync: ['--bar']
  };

  shipit.remoteCopy('src', 'dest', {
    direction: 'remoteToLocal',
    rsync: '--foo'
  });

  expect(shipit.pool.copy).to.be.calledWith('src', 'dest', {
    direction: 'remoteToLocal',
    ignores: ['foo'],
    rsync: ['--foo']
  });
});
```

**Comment by neoziro** [comment]
@grEvenX yes for sure, global are the defaults, and in method we can override them, be careful of mutability!

**Comment by grEvenX** [comment]
@mochja Can you update the PR, perhaps it'll actually get merged ;)

**Comment by mochja** [comment]
@grEvenX Yes, I can update it.

The reason why I did it this way was that it would not need additional changes to [shipit-deploy](https://github.com/shipitjs/shipit-deploy/blob/bdf0c4cd70dbff52a6bdcbf73d492786a8b81e30/tasks/deploy/update.js#L71).

How would like you to handle the new behavior now ? This approach would introduce BC break if you didn't had rsync options defined with `--del`, wouldn't it ?

**Pull request merged**